core/src/main/java/com/sequenceiq/cloudbreak/reactor/DecommissionHandler.java

```diff
@@ -67,13 +68,15 @@ public void accept(Event<DecommissionRequest> event) {
         try {
             Stack stack = stackService.getByIdWithListsInTransaction(request.getStackId());
             hostNames = getHostNamesForPrivateIds(request, stack);
-            Map<String, HostMetadata> hostsToRemove = ambariDecommissioner.collectHostsToRemove(stack, hostGroupName, hostNames);
-            Set<String> decommissionedHostNames;
-            if (skipAmbariDecomission(request, hostsToRemove)) {
-                decommissionedHostNames = hostNames;
-            } else {
-                executePreTerminationRecipes(stack, request.getHostGroupName(), hostsToRemove.keySet());
-                decommissionedHostNames = ambariDecommissioner.decommissionAmbariNodes(stack, hostsToRemove);
+            Set<String> decommissionedHostNames = hostNames;
+            if (!skipAmbariDecomission(request)) {
+                LOGGER.debug("Downscale is not forced. Collecting nodes for decommission.");
+                Map<String, HostMetadata> hostsToRemove = ambariDecommissioner.collectHostsToRemove(stack, hostGroupName, hostNames);
+                if (!CollectionUtils.isEmpty(hostsToRemove)) {
+                    LOGGER.debug("Hosts about to be removed are present in Ambari. Proceeding with decommission.");
+                    executePreTerminationRecipes(stack, request.getHostGroupName(), hostsToRemove.keySet());
```

Comment:
Are we sure pretermination recipes are not necessary in case of forced downscale or in case of there is 0 hosts to remove on ambari side?